### PR TITLE
[tooling] Add exclusions to black for auto generated code

### PR DESCRIPTION
**Pull request by kdestin**
# Description

This PR:

* Moves the configuration for `black` to a separate file (`eng/black-pyproject.toml`) to address feedback raised in #28784 
* Adds exclusions to the black config, so that it ignores some directories that should be excluded from formatting (e.g. autogenerated code, vendored code, etc...). (See motivation for `azure-ai-ml` here: https://github.com/Azure/azure-sdk-for-python/pull/28817#issuecomment-1431678918)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by scbedd — APPROVED** [review]
I am going to approve this, but let me get sign off from one of the python devs as well ok @kdestin ?

**Comment by kdestin** [comment]
> I am going to approve this, but let me get sign off from one of the python devs as well ok @kdestin ?

Works for me

File: eng/black-pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[tool.black]
+line-length = 120
+extend-exclude = '''
+# Exclude some directories from formatting (note the slashes at surrounding the pattern)
+/(
+  | _vendor
+  | _generated
```
Comment (harneetvirk) [inline eng/black-pyproject.toml:7]: Is it possible to exclude the project specific directories?

Comment (kdestin) [reply eng/black-pyproject.toml:7]: I'm unsure what the Azure SDK team's stance on that would be.

What did you have in mind specifically to exclude?

Comment (kdestin) [reply eng/black-pyproject.toml]: Removed the exclusions for `_vendor` and `_generated`, so we're probably good here.

Comment (kdestin) [reply eng/black-pyproject.toml]: Resolved by [2b42d02](https://github.com/Azure/azure-sdk-for-python/pull/28835/commits/2b42d02000544906f5d39ea97c27421c8023f748)

**Pull request merged**